### PR TITLE
Add an `Actor` custom derive macro

### LINKED ISSUE #127: Add a custom-derive for `Actor`
Although it doesn't require much boilerplate at the moment, I think it would be a nice addition to add a `derive(Actor)` macro that generates a default `Actor` implementation with `Stop = ()`.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Fixes #127.

### CODE REVIEW

File: macros-test/tests/pass/basic.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#[derive(xtra::Actor)]
+struct MyActor {
+
+}
```
Comment (thomaseizinger) [inline macros-test/tests/pass/basic.rs:4]:
```suggestion
struct MyActor;
```

File: macros-test/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "macros-test"
+version = "0.1.0"
+edition = "2021"
+publish = false
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (Restioson) [inline macros-test/Cargo.toml:7]: Small nit :)

File: macros/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "xtra-macros"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+proc-macro = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (Restioson) [inline macros/Cargo.toml:9]: same vein as first nit

**Review by Restioson — APPROVED** [review]
I like this change in concept, but I do wonder if this is the most optimal directory structure/layout. I see that https://github.com/tokio-rs/tokio has a separate directory for the main tokio crate, whereas this has the crate root as `.`. This also doesn't use a workspace, I think.

File: macros-test/Cargo.toml (hunk 2 above)
Comment (thomaseizinger) [reply macros-test/Cargo.toml:7]: You'd want this comment removed or what is the nit? :D

**Comment by thomaseizinger** [comment]
> I like this change in concept, but I do wonder if this is the most optimal directory structure/layout. I see that https://github.com/tokio-rs/tokio has a separate directory for the main tokio crate, whereas this has the crate root as `.`. This also doesn't use a workspace, I think.

We do use a workspace here, it is defined in the main crate!

I don't care much about the structure but if we want to change it, I'd suggest a different PR because it will be quite the diff noise :)

File: macros-test/Cargo.toml (hunk 2 above)
Comment (Restioson) [reply macros-test/Cargo.toml:7]: Yea I think it's best to leave it out as it's just fluff

Comment (thomaseizinger) [inline macros-test/Cargo.toml:7]:
```suggestion
```

File: macros/Cargo.toml (hunk 3 above)
Comment (thomaseizinger) [inline macros/Cargo.toml:9]:
```suggestion
```

**Comment by Restioson** [comment]
> We do use a workspace here, it is defined in the main crate!

Oh, I hadn't seen that!

**Pull request merged**